temporal-sdk/src/main/java/io/temporal/internal/client/RootWorkflowClientInvoker.java

```diff
@@ -200,6 +170,151 @@ public WorkflowSignalWithStartOutput signalWithStart(WorkflowSignalWithStartInpu
+            WorkflowExecution.newBuilder().setWorkflowId(startInput.getWorkflowId()).build(),
+            Optional.of(startInput.getWorkflowType()),
+            updateOptions.getUpdateName(),
+            Header.empty(),
+            updateOptions.getUpdateId(),
+            updateOperation.getUpdateArgs(),
+            updateOptions.getResultClass(),
+            updateOptions.getResultType(),
+            "", // firstExecutionRunId is always empty
+            WaitPolicy.newBuilder()
+                .setLifecycleStage(updateOptions.getWaitForStage().getProto())
+                .build());
+    UpdateWorkflowExecutionRequest updateRequest =
+        toUpdateWorkflowExecutionRequest(updateInput, dataConverterWithWorkflowContext);
+
+    ExecuteMultiOperationRequest request =
+        ExecuteMultiOperationRequest.newBuilder()
+            .setNamespace(clientOptions.getNamespace())
+            .addOperations(
+                0,
+                ExecuteMultiOperationRequest.Operation.newBuilder()
+                    .setStartWorkflow(startRequest)
+                    .build())
+            .addOperations(
+                1,
+                ExecuteMultiOperationRequest.Operation.newBuilder()
+                    .setUpdateWorkflow(updateRequest)
+                    .build())
+            .build();
+
+    ExecuteMultiOperationResponse response;
+    try {
+      response = genericClient.executeMultiOperation(request);
+    } catch (StatusRuntimeException err) {
+      MultiOperationExecutionFailure failure =
+          StatusUtils.getFailure(err, MultiOperationExecutionFailure.class);
+      if (failure == null) {
+        throw err;
+      }
+
+      if (failure.getStatusesCount() != request.getOperationsCount()) {
+        throw new Error(
+            "Server sent back an invalid error response: received "
+                + failure.getStatusesCount()
+                + " instead of "
+                + request.getOperationsCount()
+                + " operation errors");
+      }
+
+      MultiOperationExecutionFailure.OperationStatus startStatus = failure.getStatuses(0);
+      if (startStatus.getDetailsCount() == 0
+          || !startStatus.getDetails(0).is(MultiOperationExecutionAborted.class)) {
+        throw Status.fromCodeValue(startStatus.getCode())
+            .withDescription(startStatus.getMessage())
+            .asRuntimeException();
+      }
+
+      MultiOperationExecutionFailure.OperationStatus updateStatus = failure.getStatuses(1);
+      if (updateStatus.getDetailsCount() == 0
+          || !updateStatus.getDetails(0).is(MultiOperationExecutionAborted.class)) {
+        throw Status.fromCodeValue(updateStatus.getCode())
+            .withDescription("Invalid StartOperation: " + updateStatus.getMessage())
+            .asRuntimeException();
+      }
+
+      throw err; // no detailed failure was found
+    }
+
+    if (response.getResponsesCount() != request.getOperationsCount()) {
+      throw new Error(
```

Comment:
We should not throw `Error` from the workflow client. The SDK never throws  `Error` from the client, generally `Error` should only be thrown if there is something critically wrong and unrecoverable within the application itself, not some downstream service. (In Workflow code we throw errors as well, but that is to avoid users catching internal exceptions and is more an exception then the rule)